services/dao-service.ts

```diff
@@ -0,0 +1,155 @@
+import { 
+  Network_v2,
+  OraclesResume,
+  Web3Connection
+} from "@taikai/dappkit";
+import getConfig from "next/config";
+
+import { NetworkParameters } from "types/dappkit";
+
+const { publicRuntimeConfig } = getConfig();
+
+export default class DAO {
+  private _web3Connection: Web3Connection;
+  private _network: Network_v2;
+
+  get web3Connection() { return this._web3Connection; }
+  get network() { return this._network; }
+
+  constructor() {
+    if (!publicRuntimeConfig?.web3ProviderConnection) 
+      throw new Error("Missing web3ProviderConnection in next.config.js");
+
+    this._web3Connection = new Web3Connection({
```

Comment:
you can have this prop as a `readonly` and assign value to it without a constructor, 

```
readonly web3Connection = new Web3Connection({ /* ... */});

constructor() {
    if (!publicRuntimeConfig?.web3ProviderConnection) 
      throw new Error("Missing web3ProviderConnection in next.config.js");
}
```

typescript will essentially do what you're doing, but provide "web3connection" as writable. Since we're using Typescript _we_ won't be able to change that value through code or lint will scream bloody murder -- and that's essentially what we want; we don't need _an actual readonly value_ because no one, but us, is consuming this service ;)

ps, [proof](https://www.typescriptlang.org/play?#code/DYUwLgBAZghglsCBeaNgGcQG4BQODGwM66EAghAN44S0QBOIMAJgPYB2wAnuchAIy46EfB3Rh6AV3xhW9ABQBKKjWF04UCPNgJFqtXTAALeqwDuEdiAsBReqYUByHaGaPFQugF8cPnAHp-CDMERAA5AHkAFVQEAjFIGD4rCzIlXFF2dFZQADpgVgBzeRhcsmVAgTwXPglJbBwJHmphTPEIACNk63J03xEYMHwjLRBlFro2nJB8ovlHAGEyAFUAcQAJGIAuABF3T1opvILisawISvxB4YhZCABbGABrEAh0SUYAQl8gA)